cpp/src/arrow/util/crc32.cc

```diff
@@ -0,0 +1,961 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Could you put this to `cpp/src/arrow/vendored/crc32.c` (or `crc32.cc`?) like `cpp/src/arrow/vendored/base64.cpp`?
We can update vendored code easily by reusing the original code as much as possible.

Could you also add an entry for `crc32.c` to `LICENSE.txt`?

Reply:
Could you also add the original source information such as the repository URL and revision?
See also `**/README.md` under `cpp/src/arrow/vendored/`.

Reply:
Will do it tonight

Reply:
> Could you also add the original source information such as the repository URL and revision? See also `**/README.md` under `cpp/src/arrow/vendored/`.

But the way, I did some tiny updates, like change it's macro and functions to arrow's macro and functions, like byte-swap, endian.

Reply:
Could you add the information to the `README.md` too?
`cpp/src/arrow/vendoered/crc32/crc32.c` (or `crc32.cc`) and `cpp/src/arrow/vendored/crc32/README.md` will be created.

Reply:
If non-trivial changes were made to the code then it should stay in `arrow/util`, IMHO. Code in `arrow/vendored` should be mostly identical to upstream and easily updateable.